\documentclass[12pt,a4paper]{article} % document setting, articles do not allow chapters
\usepackage[english]{babel} % english spell-check
\usepackage[utf8]{inputenc} % allows the input of special characters from keyboard
\usepackage[T1]{fontenc}
\usepackage{lmodern} %load modern fonts
\usepackage{helvet} % lead helvetica font
\renewcommand{\familydefault}{\sfdefault} % without this helvetica does not work on linux
\usepackage{textcomp} % for text symbols such as (C)
\usepackage{siunitx} % for mathematical embedding with $

\AtBeginDocument{\renewcommand{\bibname}{References}} % change Bibliography to Reference in TOC
\usepackage[compress]{natbib} % include bibliography
%super, numbers
\bibliographystyle{apalike} % simple bibliography style % sudo texhash
%\bibliographystyle{plainnat} %order by appearance: unsrtnat
\usepackage{hyperref} % allow printing URLs (here: for Bibliography)
\usepackage{cite}

\usepackage{graphicx} % include figures
\usepackage[labelfont=bf]{caption} % change figure caption to bold

\usepackage{titlesec} %change title style
\titleformat*{\section}{\normalsize\bfseries}
\titleformat*{\subsection}{\normalsize\itshape}

%\usepackage[mathlines]{lineno} % add line numbers
\usepackage{setspace} %for double spacing

\usepackage{booktabs}
\usepackage{makecell}

\author{Masumi Stadler}
\title{PhD Manuscript 1}

%%%%%%%%%%%%
% Document %
%%%%%%%%%%%%
\begin{document}
%\linenumbers

\setlength{\parindent}{0cm}
Title (max 150 char): Diverging DNA and RNA communities along a boreal terrestrial-hydrological continuum (?) \\

Masumi Stadler\textsuperscript{1,*}, Paul A. del Giorgio\textsuperscript{1}\\

Affilitation:\\
(1) Groupe de Recherche Interuniversitaire en Limnologie (GRIL), D\'{e}partement des Sciences Biologiques, Universit\'{e} du Qu\'{e}bec \`{a} Montr\'{e}al, Montr\'{e}al, QC, Canada


Institutional e-mail addresses: MS (stadler.masumi@courrier.uqam.ca), PdG (del\_giorgio.paul@uqam.ca)\\

Running title (max 50 char): Microbial communities along a boreal terrestrial-hydrological continuum.\\

*Contact corresponding author: Masumi Stadler \\
E-mail: m.stadler.jp.at@gmail.com \\
Phone: +1 (514)297-5330 \\
Address: D\'{e}partement des Sciences Biologiques, Universit\'{e} du Qu\'{e}bec \`{a} Montr\'{e}al, Case Postale 8888, Succursale Centre-Ville, Montr\'{e}al, QC, H3C 3P8, Canada \\

Keywords: aquatic bacterial communities, terrestrial-aquatic continuum, ecosystem connectivity, residence time, boreal ecosystems, mass effects, environmental sorting, metacommunity, rare biosphere\\

Author contributions: PdG designed the sampling, MS collected the data, MS analysed the data, MS and PdG discussed the results and wrote the manuscript.\\

\newpage

\doublespacing

%\begin{abstract}


%\end{abstract}


\setlength{\parindent}{1cm}
\section*{Introduction}
Since the 1880s, we are starting to grasp the intriguing variety of microbial physiological pathways and how they are involved in many elemental elemental cycling processes on a global scale \citep{Caumette2015}. The variety in physiological pathways has been hypothesised to arise from the observed vast taxonomic diversity. Microbial communities found outside of the petri-dish are inherently taxonomically diverse, with richness estimations that defy imagination \citep{Thompson2017, Louca2019}. Richness varies among ecosystems with highest taxonomic reservoirs found in soil and less so in comparably nutrient depleted aquatic ecosystems \citep{Thompson2017}. While descriptive approaches of richness and taxonomic composition is common in almost any environment imaginable, little do we actually understand how microbes assemble into these diverse communities \citep{Shade2017a, Shade2018}. Assembly processes have been defined into four fundamental processes - selection, dispersal, diversification and drift - \citep{Vellend2010}, which can be characterised by their degree of determinism and stochasticity \citep{Zhou2017}. Recent popular statistical approaches aim to quantify each assembly process, however, it relies on an assumption that taxonomically related organisms have similar functional traits \citep{Stegen2013a, Stegen2015a}. Phylogenetic conservatism is a concept that has been handled with care within the macro-biologies \citep{Losos2008, Warren2008}, but should be a target of even stronger discussions in microorganisms. Indeed, many microbial traits have been shown to be phylogenetically conserved \citep{Martiny2015} but it was also noted that more complex traits encoded by many genes are more likely to be conserved and simpler traits that involve fewer genes are rather phylogenetically dispersed \citep{Martiny2013a}. It is also na\"{i}ve to assume that traits harboured in the genome are exhibited by default (xx) and thus are the only traits a microbe can express (i.e,  epigenetics (xx)). The increasing literature on potential niche segregation within the same species (i.e. ecotypes) exemplify intraspecies versatility that can blur patterns based on phylogenetic conservatism \citep{Achtman2008, Ackermann2015, Chase2018}.

In concert with the lack of suitable approaches, traditionally, microbial community dynamics have been examined one ecosystem at a time. Most field studies, investigate the intriguing seasonal and spatial fluctuations of community composition within individual terrestrial, freshwater and marine ecosystems (hereafter ecosystem domains) \citep{Shigyo2019,Jones2012,Hassell2018,Giovannoni2012}. Even among freshwater studies, researchers tend to focus on either streams, lakes, or rivers \citep{Logue2008}. While this approach gives an in-detail insight into ecosystem specific community dynamics, it neglects an inherent characteristic of nature: connectivity beyond ecosystem boundaries.

Literature is scarce when searching for studies that cover multiple ecosystem domains within a study \citep{Nemergut2011, Shade2013} and it is even rarer to investigate multiple ecosystem domains that are actually physically connected \citep{Ruiz-Gonzalez2015, Hauptmann2016, Doherty2017}. Although microorganisms without motility machineries are seemingly stationary, the miniscule size of microorganisms promotes dispersal. Although dispersal limitation has been observed even for microbes \citep{Hanson2012}, within a watershed freshwaters act as a carrier of matter, from the terrestrial milieu over freshwater networks to ultimately the ocean. Especially during strong rain events, freshwater flushes through the earth, extracting all soluble nutrients and capturing matter - dead or alive - along its hydrological evolution from a raindrop to collectively becoming a stream. On its further journey, the rushing water may be stopped by lakes and reservoirs that temporarily give more time for some organisms to thrive and previously unavailable resources may be degraded \citep{Catalan2016a}. Within the aquatic network, dynamics in hydrology have been determined to be one of the major drivers of microbial community composition \citep{Nino-Garcia2016} and evidence suggests that a large proportion of aquatic microbial communities can be traced back to soils \citep{Crump2012, Besemer2013, Ruiz-Gonzalez2015, Hauptmann2016}. Different branches of the aquatic network collect taxa from all sub-catchments within a watershed, potentially creating sub-watershed specific unique communities at each sub-catchment's mouth. Thus, not every member within diverse communities at a single location shares a similar history in terms of origin \citep{Nino-Garcia2016, Comte2017} or reacts similarly to changing environmental conditions \citep{Fierer2007}. However, such passive dispersal implies that occurence of a taxon in an unsuitable habitat merely by accident is indeed likely.

Death and dormancy \citep{Cole1999, Jones2010} questions the suitability of DNA approaches to study ecological 'communities'. In a strict ecological sense, a community refers to an assemblage of various species that interact with each other, sharing niches in forms of available space or resources \citep{Konopka2009}. However, DNA methods are not able to distinguish the actively participating from the passive members. Although being target to debate, researchers have expanded their molecular toolbox to RNA approaches to capture taxa that have invested in protein synthesis in the recent minutes to hours \citep{Blazewicz2013}. Evidence of 16S rRNA has opened windows to study differential responses of the active community to environmental variables and available resources \citep{Osterholz2016}. To this date, the full potential of RNA approaches in understanding microbial community assembly and dynamics across ecosystems has yet to be explored.

In this study, we followed the Romaine river in North-Eastern Qu\'{e}bec, Canada, as an example watershed over several years and seasons. Our overall objective was to understand microbial succession and assembly processes along a terrestrial-hydrological continuum. Therefore, soils, soilwater, groundwater, streams, the river, lakes, reservoirs and the estuary were sampled for DNA and RNA and sequenced for the 16S rRNA gene to characterise the taxonomic composition of each fraction. We specifically wanted to explore firstly, whether we have distinct microbial communities depending on habitat types and if we can find seasonality across all ecosystems. We hypothesised that firstly, seasonality will be reflected in all communities of the different ecosystems as boreal climate zones do exhibit strong seasonality that is accompanied by shifts in available light, temperature and thus affect vegetation and aquatic water bodies substantially. Secondly, we expected a gradual change in community composition along the continuum rather than distinct separate clustering of terrestrial and aquatic communities. Thirdly, we hypothesised that dynamics in DNA and RNA assemblage divergence and convergence enables us to grasp what dominating assembly process (i.e. mass effect and selection) is governing the community at a certain point in the continuum. And lastly, we wanted to explore where in a typical rank abundance curve (e.g. rare vs abundant taxa) community reshuffling is happening as the communities change habitats along the continuum.
%Furthermore, the consecutive flooding of three reservoirs over the sampled years created a unique opportunity to study the transformation process of a running system into a standing water body.
%(should be around 900~1000 words)

\section*{Material and methods}
\subsection*{Catchment characteristics and sampling design}
To follow the movement of microbial communities within a watershed, samples were taken along the Romaine river (C\^{o}te-Nord region, Qu\'{e}bec, Canada) (Fig. 1a-b) for three years from 2015-2017. The Romaine catchment belongs to the eastern black spruce-moss bioclimatic domain and drains an area (\textit{A}) of approximately 14 500 km\textsuperscript{2}. The catchment was glaciated 7 000 – 10 000 years ago and left mostly a till blanket and veneer as surficial material. It is mainly dominated by acid rocks (e.g. granodiorite, granite, quart diorite) with granitised sedimentary and volcanic rock, and has isolated patches of permafrost (0-10\%)(Natural Resources Canada). The soil is composed of roughly 61.4\% sand, 31.9\% silt, 6.7\% clay and stores approximately 140.4 t ha\textsuperscript{-1} of organic carbon (in top 5 cm; given are catchment averages)\citep{Lehner2013, Hengl2014}.

The river springs between the Atlantic and Saint Lawrence watersheds (52\textdegree{}52'20"N 63\textdegree{}36'55"W; 702 masl), and consequently flows through a series of lakes (hereafter riverine lakes) including the biggest lake in the catchment – Lake Br\^{u}l\'{e} (\textit{A}: 127.11 km\textsuperscript{2}, elevation: 470 masl). The river mainly flows towards the South with a maximum distance from the northern headwaters to the river mouth expanding to approximately 475.1 km. Nearly half of the catchment is covered by coniferous forests (\textit{P.mariana}-moss), with mixed forests being rather minor (11\%) and deciduous stands with white birch (\textit{Betula papyrifera}) and trembling aspen (\textit{Populus tremuloides}) are even more rare (2\%)\citep{HQreport2009}. The northern part of the catchment is characterised by a flat open black spruce (\textit{Picea mariana})-lichen forest with shrubs and moss-lichen (Fig. S1a). As one follows the river downstream, the relief changes drastically to a steep mountainous stretch that forms sections of canyons (Fig. S1b). The river looses 330 m of elevation from the mountainous section until it makes a sharp turn to the west into the lower coastal plain. The coastal plain is characterised by peatland areas with swamps and shallow waters that are completely permafrost free (Fig. S1c). There are two larger tributaries in the coastal plain that flow through the lakes Puyjalon (\textit{A}: 13.10 km\textsuperscript{2}) and Allard (\textit{A}: 19.24 km\textsuperscript{2}). A weather station located in the lower coastal plain (50\textdegree{} 16'55.000" N, 63\textdegree{} 36'41.000" W, Havre-Saint-Pierre Airport, Natural Resources Canada) recorded an annual precipitation of 810.77 $\pm$ 35.25 mm and 1.18 $\pm$ 0.73 \textdegree{}C, -32.63 $\pm$ 1.36 \textdegree{}C, and 25.8 $\pm$ 0.66 \textdegree{}C for mean, minimum and maximum temperature over the sampled years.

The Romaine river has been dammed during the sampling period, forming a reservoir cascade complex with 4 reservoirs by 2020. The reservoirs Romaine 2 (RO2, \textit{A}: 81.15 km\textsuperscript{2}, mean depth 61 m), Romaine 1 (RO1, \textit{A}: 13.22 km\textsuperscript{2}, mean depth 22 m) and Romaine 3 (RO3, \textit{A}: 35.18 km\textsuperscript{2}, mean depth 66 m) were flooded in the years 2014 (winter), 2015 (winter), and 2017 (spring), respectively.

Overall, 395 samples were collected for DNA (D) and 202 for RNA (R), covering spring (166-D, 69-R), summer (195-D, 99-R) and autumn (34-D, 34-R). RNA samples were sampled from 2016 onwards. To follow a terrestrial-aquatic continuum various habitat types have been sampled (Table S1). Due to the remoteness and inaccessibility of the northern most headwaters, we have sampled the Petite Romaine sub-catchment (PR, \textit{A}: 310.73 km\textsuperscript{2}, elevation: 580 masl, Fig. 1c) for streams and headwater ponds. This sub-catchment represents a headwater stream network in our studied continuum.

Surface soil samples were collected by mixing three randomly selected cores (30 cm) that were taken in proximity of installed piezometers to sample soil-water. The upper 5 cm including surface vegetation were removed before the soil was transferred into a sterile plastic bag. Three piezometers were randomly installed in proximity (30-100 cm) to a sampled stream with an average depth of 50 $\pm$ 20 cm. However, if the piezometers were installed too close to the stream main channel, hyporheic water was sampled instead. Piezometers were emptied 3 times (1-2 h) with a peristaltic pump before sample water was collected. The water from the randomly installed piezometers were pooled together. Groundwater was direcy collected from constructed wells with submersible pumps. Surface water samples were directly collected into a pre-rinsed carboy bottle at a depth of 0.5 m, close to the shore for stream samples and diverse locations within the river and reservoirs. Lake sediment samples were collected with sediment cores (1-2 m depth), and the upper 10 cm were collected and mixed for subsequent processing. All samples were stored at 4 \textdegree{}C upon arrival at the laboratory until further processing on the same day of sampling.

\subsection*{Sample processing and sequencing}
Homogenised soil and sediment samples were transferred to aliquots of 0.25 g and frozen immediately on the same day of sampling. A minimum of 25 mL and 250 mL of soil-/hyporheic-water and surface water was filtered through 0.22 \si\micro m polycarbonate membrane filters (Merck Millipore, Darmstadt, Germany), respectively, and subsequently stored frozen. All DNA and RNA samples were frozen at -20 \textdegree{}C at the field station and further stored at -80 \textdegree{}C at the university laboratory until extraction. Samples for RNA extraction were submerged in RNAlater\textsuperscript{\textregistered} and LifeGuard\textsuperscript{\textregistered} Soil Preservation solution (QIAGEN\textsuperscript{\textregistered}, Hilden, Germany) for water and humic samples (soil, soilwater, hyporheic water), respectively. To allow stabilisation, samples were left at 4 \textdegree{}C overnight and were subsequently stored frozen.

Following the manufacturers instructions, PowerWater\textsuperscript{\textregistered} and PowerSoil\textsuperscript{\textregistered} DNA and RNA extraction kits (MoBio, Carlsbad, CA, USA) were used to extract water and soil/soil-/hyporheic-water/sediment samples, respectively. In 2017, the equivalent DNeasy\textsuperscript{\textregistered} and RNeasy\textsuperscript{\textregistered} PowerWater\textsuperscript{\textregistered} Kits (QIAGEN\textsuperscript{\textregistered}) were used for DNA and RNA samples, respectively. cDNA was synthesised from the extracted RNA with a high capacity cDNA Reverse Transcription Kit (Applied Biosystems\textsuperscript{\texttrademark}, Foster City, CA, USA). Successful extraction was evaluated via PCR amplification of the 515F-806R primers (IDT Technologies, Coralville, IA, USA) and DNA concentration was measured with a NanoDrop 2000c (Thermo Fisher Scientific Inc., Waltham, MA, USA). Extracts were sent to G\'{e}nome Qu\'{e}bec Innovation Centre (Montr\'{e}al, QC, Canada) for paired-end metabarcoding of the 16S rRNA V4 region using the primers 515F (5'-GTGCCAGCMGCCGCGGTAA-3') and 806R (5'-GGACTACHVGGGTWTCTAAT-3') on a Illumina MiSeq (PE250) platform.

\subsection*{Bioinformatic analysis}
Primers were removed from 16S rRNA gene and rRNA data sets using the software \textit{cutadapt} (Version 1.18, \citet{Martin2013}), which allows for the removal of the primer sequence and its variants in their true and complement orientations. Additionally, all reads shorter than 125 nucleotides were removed as they cannot achieve a minimum overlap necessary for paired-end merging in downstream processing.

To identify amplicon sequence variants (ASVs), 16S rRNA amplicon reads were analysed through the DADA2 (Divisive Amplicon Denoising Algorithm 2) pipeline (Version 1.14.1, \citet{Callahan2017}) on R Version 3.6.3 \citep{RCoreTeam2017}). Read qualities were evaluated for each sequencing plate separately and read length was trimmed according to their quality scores. The most conservative trimming criteria still allowed for an overlap of forward and reverse reads by 60 bp. Samples were pooled by plate, season and sequencing depth for learning the error rates. DADA2 runs on a sample by sample basis, and thus removes observed singletons by sample to avoid inclusion of false-positive sequencing errors. To retain more rare taxa within a sampling campaign (year-season combinations) along the continuum, samples were 'pseudo'-pooled for the \textit{dada()} step.  This step enables the removal of singletons by pool but retains singletons within a sample. Paired-ends were merged after successful inference of amplicon variants. Subsequently, all ASVs that are identical in sequence but differ only by length were merged together with \textit{collapseNoMismatch()}, leading to ASVs that represent a unit similar to 100\% similarity operational taxonomix units (OTUs) and chimeras were removed (\textit{removeBimeraDenovo()}). Finally, taxonomy was assigned with the \textit{DECIPHER} package (Version 2.14.0, \citet{Wright2016}) implementing the increased accuracy IDTAXA algorithm \citep{Murali2018} and the provided trained classifier of the SILVA database (Version 138, \citet{Pruesse2007}). Only ASVs that were classified as Bacteria and not as Mitochondria or Chloroplast were evaluated in this study. Even after merging ASVs into 100\% OTUs, several ASVs were found to be highly abundant only in RNA. To account for slight differences that may have emerged between DNA and RNA ASVs and also to merge potential differences among 16S rRNA copies within a single genome \citep{Vetrovsky2013a}, ASVs were merged into OTUs by a 99\% similarity threshold (\textit{DECIPHER}: \textit{AlignSeqs(), DistanceMatrix(), IdClusters()}, \citet{Wright2016}). To keep computational power low, OTU clustering was conducted within taxonomical pools. Such that ASVs classified only as "Bacteria" without any classification in further ranks were pooled for clustering, while any ASV with a classifcation in any higher rank was clustered with other ASVs that had the same highest assigned taxonomical rank (e.g. "Bacteria-Proteobacteria-Gammaproteobacteria-Burkholderiales-Burkholderiaceae-Limnobacter" pool).

All OTU observations with less than 10 reads per sample were removed. Furthermore, \textit{metagenomeSeq} was used to transform and stabilise variation in library sizes with cumulative sum scaling (CSS) \citep{Paulson2013}. CSS results were rounded to its integer to represent count data (hereafter: CSS reads). CSS results were compared with results achieved with various rarefaction thresholds. To account for random sampling effects, rarefaction was conducted with 100 random iterations. Similarly to the CSS data treatments, the mean of all iterations was rounded to its integer to be used for subsequent analysis. There were no substantial differences in the results between CSS and various rarefaction thresholds (Fig. S2). Consequently, CSS results were used for downstream analyses.

\subsection*{Data exploration and statistical analyses}
To explore differences in microbial community composition across habitat types and seasons, a Principal Coordinates Analysis (PCoA) was conducted with Bray-Curtis dissimilarities ($D_{BC}$) \citep{Bray1957, Legendre1998} based on all DNA samples with the function \textit{pcoa} in the \textit{ape} package \citep{Paradis2018}) (n = 372, 20185 OTUs). The community matrix was log transformed, (log\textsubscript{2}(x + 1)) to resolve a strong horse-shoe effect. The $D_{BC}$ matrix was square-root transformed to make the distance matrix Euclidean \citep{Legendre1998, Borcard2011}. A PERMANOVA was computed with 9999 permutations to evaluate statistical differences in habitat type and season  with the \textit{adonis} function in \textit{vegan} \citep{Oksanen2017}. The null hypothesis of this approach is that the cluster centroids do not differ between groups \citep{Anderson2013}. The test cannot distinguish among-group from within-group variation if data disperion is variable among groups. Thus, an analysis of multivariate homogeneity was computed with \textit{betadisper} in \textit{vegan} \citep{Oksanen2017} by accounting for small and uneven sample sizes with the \textit{bias.adjust} argument. Using \textit{permutest} in \textit{vegan} \citep{Oksanen2017}, we tested whether dispersion is variable among groups. The null hypothesis is that the average within-group dispersion is equivalent in all groups \citep{Anderson2013}.

Secondly, to evaluate whether sampled RNA communities were further different from the DNA assemblages, we performed a second PCoA ($D_{BC}$ with square-root transformation) with both DNA and RNA samples (n = 572, 20185 OTUs). Again, statistically different clusters were investigated with a PERMANOVA (9999 permutations), where habitat type, season and nucleic acid type (DNA vs. RNA) formed the clusters. The same framework explained above to check for dispersions was applied. For all statistical analyses, an $\alpha$ level of 0.05 was chosen prior to analysis.

To examine how different DNA and RNA assemblages of the same sample are, the distance of each DNA-RNA sample pair within the PCoA ordination space was computed. Distance was calculated along the first three axes, as these axes captured clear ecosystem and temporal patterns:

\[ m = \sqrt{(\mid x_{DNA} - x_{RNA} \mid)^2 + (\mid y_{DNA} - y_{RNA} \mid)^2 + (\mid z_{DNA} - z_{RNA} \mid)^2}\]

where $x$, $y$ and $z$ are site scores on the first three axes. This distance metric extracts the variation within pair-wise dissimilarities that are explained by the factors driving the first three axes. As a proof of concept, summation of the distances across all PCoA dimensions (571 axes) equals the initial pair-wise Bray-Curtis dissimilarites (Fig. S3). Thus, calculating the distance across a subset of axes extracts the variance of dissimilarities explained by the selected axes.

To further evaluate when sample differences are driven by abundance differences and when by species sorting, we computed a PCoA with the S{\o}rensen dissimilarity ($D_{S}$), which is the incidence based equivalent of $D_{BC}$ (square-root transformed to achieve Euclidean space) \citep{Sorensen1948, Legendre1998}. We further applied the same framework of calculating the distance among DNA and RNA pairs and computed a ratio between the S{\o}rensen ($m_S$) and Bray-Curtis distance ($m_{BC}$) to see when incidence based dissimilarity is greater than abundance and vice versa. This framework allows us to understand what is causing DNA and RNA assemblage divergence: shifts in abundance or species sorting.

Non-linear regressions were computed to test the hypothesis that DNA-RNA distances are a function of $\alpha$ diversity indices (Shannon-Wiener index and Pielou's evenness). Regressions were computed on the means of habitat types. The best fit of the relationship was evaluated by computing linear regressions and polynomials of second and third degrees. The best model was selected based on a higher adjusted R\textsuperscript{2} and low residual sum squares.

\subsubsection*{Abundance classification}
Traditionally, abundance groups (AGs) such as "abundant" and "rare" have been defined by various relative abundance thresholds ranging from 0.1 to 1 \% within the literature. While inconsistencies hinder comparisons among studies, we additionally are working with variance stabilised read numbers, thus traditional thresholds based on relative abundances are not applicable. In order to classify OTUs into AGs, we developed a new framework to classify OTUs into AGs based on the shape of rank abundance curves of each habitat. We initiate the framework by calculating the mean abundance of each OTU by habitat type. Subsequently, for each habitat type a smoothed rank abundance curve was generated with the function \textit{smooth.spline} with the smoothing parameter set to 0.7 in base R (Fig. S4a). Ranks that correspond to moments of acceleration along the curve were identified by taking the second derivative of the log(x + 1) transformed abundance curve (Fig. S4c). All OTUs ranked below the second maximum acceleration were defined as rare. OTUs falling above the first maximum acceleration were defined as abundant, while the section in between the two maxima represents moderately abundant OTUs (Fig. S2). The CSS reads corresponding to the ranks identified for AG were extracted for each habitat type seperately. Subsequently, the average CSS reads for each abundance threshold was calculated. This approach classified all OTUs with >= 47 CSS reads as abundant, < 47 and >= 5 CSS reads as moderate, and < 5 CSS reads as rare.

As the second step, we defined OTUs into spatial abundance groups (spAGs) based on their DNA abundance in each sample (Table 1). In brief, we distinguish OTUs that are present everywhere (universal [AG], cosmopolitan) from those that can be absent in certain habitat types. Additionally, there are categories for OTUs with different abundance (abundant, moderate, rare) and whether they change AGs between habitats (shifters).

To evaluate how spAGs are distributed among the habitat types, we computed weighted averages scores of each OTU for the DNA-RNA PCoA with the \textit{wascores} function in \textit{vegan} \citep{Oksanen2017}. These species scores represent optima of each OTU within the PCoA ordination. Furthermore, we calculated the difference in abundance between DNA and RNA for each OTU ($\Delta$ OTU CSS reads) to explore what spAGs are causing the difference in DNA and RNA that we can observe within the PCoA. \\[.3cm]

The packages \textit{phyloseq}, \textit{tidyverse}, \textit{plyr} and \textit{data.table} were used for data wrangling and transformation \citep{McMurdie2013,Wickham2019, Wickham2011, Dowle2019}, and \textit{doMC} and \textit{parallel} enabled parallel processing \citep{Analytics2019, RCoreTeam2017}. \textit{ggplot2}, \textit{ggpubr}, \textit{ggnewscale} and \textit{cowplot} were used to visualise the results \citep{Wickham2016, Kassambara2018, Campitelli2020, Wilke2019}. All analyses have been conducted in R v3.4.2 \citep{RCoreTeam2017} and RStudio v1.3.1073 \citep{RStudioTeam2016}. Maps were created with QGIS (version 3.12) and a digital elevation model provided by Natural Resources Canada. Watersheds were delineated with ArcMap (version 10.5.1, ESRI Inc., Redland, CA) and the Spatial Analyst Toolbox.

\section*{Results}
Sampled sites covered a large range of habitat types from soils, soilwater, over streams, the main river, lakes, reservoirs and the estuary. We recovered 87 002 470 quality reads, with 156 347 identified ASVs. After sub-sampling and 99 \% similarity clustering, 47 402 282 reads and 20 220 OTUs were retained. The smallest and largest library size were both found in riverine lakes with 2851 and 1 333 391 reads, respectively. On average, the lowest library sizes were found in sediments, soil and soilwater with less than 30 000 reads. Compared to most freshwater samples that had a library size higher than 50 000 reads (Fig. S5). 33 phyla, 80 classes,  189 orders, 400 families, and 696 genera were represented in the dataset. Relative abundances of phyla varied across habitat types (Fig. S5) but on average, the community was composed of 52.11 \% Proteobacteria, 17.17 \% Actinobacteria, 8.06 \% Verrucomicrobiota, 6.88\% Bacteroidota, 2.91 \% Acidobacteriota, 2.63 \% Planctomycetota, 2.37 \% Cyanobacteria, 1.43 \% Myxococcota, 1.23 \% Desulfobacterota, 1.22 \% Chloroflexi and 1.19 \% Nitrospirota.

\subsection*{Composition of DNA assemblages along a terrestrial-aquatic continuum}
We observed a gradual separation of terrestrially-influenced habitats such as soil, soilwater and sediment from freshwater and estuary samples along the first PCoA axis capturing 14.64 \% of the variance (Fig.2a). This observation was supported by the PERMANOVA analysis (\textit{F}\textsubscript{12} = 17.09, \textit{R}\textsuperscript{2} = 0.35, \textit{p} = 0.0001). Streams, groundwater, tributaries, headwater ponds, and lakes were creating a gradient between the two very distinct clusters of terrestrial and riverine/reservoir samples (Fig.2a).

The trajectory along the continuum has a striking seasonality where spring and summer/autumn samples cluster most pronounced in the reservoir samples along the second PCoA axis capturing 3.6 \% of the variance (PERMANOVA: \textit{F}\textsubscript{2} = 11.08, \textit{R}\textsuperscript{2} = 0.038, \textit{p} = 0.0001). Soil, sediment, soilwater and groundwater samples, however, do not exhibit a clear seasonality (Fig.2b). Seasonality does not emerge as a strong driver even within a PCoA performed only with terrestrial samples (Fig.S6). It is striking how seasonality takes effect as soon as we enter the streams and headwater ponds, where spring samples gradually form a path towards the spring riverine and reservoir cluster. In contrast to spring, streams do diverge faster from the terrestrial cluster and mingle more with other freshwater samples in summer and autumn. Interestingly, lakes are located in general more within the summer/autumn trajectory. These lakes, however, are not part of the continuum but were sampled across the watershed and thus are rather a part of the extended meta-community. Similarly, sediment samples cluster in the summer/autumn path without a single case lying in the spring trajectory.

A unique characteristic of the studied continuum is the addition of reservoirs over the sampled years. While the river fraction upstream of the reservoir (Upriver) has the same upstream history over the years, the continuum that feeds into the downstream of the reservoir (Downriver) changes over the years. In 2015, the downriver sites capture water from one reservoir (RO1), while in 2016 they feed from two reservoirs (+RO2, Fig. 1c). While in summer, there is no clear seggregation, we can identify two distinct clusters in spring. Upriver and downriver sites are clustered seperately in 2016, while in 2015 the furthest downriver sites closer to the river mouth converge back to the upriver sites (Fig. 2c, 2017 spring was not sampled).

Finally, estuary samples exhibit similar seasonality as reservoirs and rivers, however, the further one moves away from the river mouth, the estuary community gradually converges back along the first and second PCoA axis towards the centre of the ordination (Fig. 2d). At the furthest site from the river mouth, the estuary community in spring seems to share community members with stream and headwater pond samples in summer.

Although PERMANOVA results strongly supported habitat type and seasonal clustering, the results could be affected by different dispersion of data. Differential levels of dispersion was found by habitat type alone (PERMDISP: \textit{F}\textsubscript{12} = 32.17, \textit{p} = 0.0001), by solely season (PERMDISP: \textit{F}\textsubscript{2} = 42.79, \textit{p} = 0.0001) and by both habitat and season combined (PERMDISP: \textit{F}\textsubscript{27} = 17.77, \textit{p} = 0.0001). Although non-homogeneity of dispersion confounds a clear conclusion based on statistical evidence that habitat type and seasons are major drivers, ordination results indicate a very apparent difference in habitat type and seasonal microbial communities. It is also a interesting finding, that dispersions vary among habitat types and seasons. While disperion between spring and summer was not statistically different (Tukey HSD: p > 0.05), they were always different when comparing autumn with other seasons (Tukey HSD: p < 0.0001). The average distance of samples within the autumn cluster to its median was smaller compared to other seasons (0.48 vs 0.62). Smaller variation in autumn likely emerges due to the fact that autumn was only sampled in one year, compared to spring and summer being sampled at least in two years. For habitat types, strongly significant differences were most pronounced between soil/soilwater/stream samples with reservoir and river samples (Tukey HSD: p = 0). In general, within habitat type dispersion was larger in terrestrial samples (PERMDISP mean avgerage distance to median: 0.6032), compared to riverine (0.5033) and reservoir (0.5124) samples. Indicating that terrestrial samples are more variable than freshwater samples. Within freshwater samples, there is a gradient in variance within stream, headwater ponds, lake and tributary clusters having highest dispersion levels close to terrestrial samples. Riverine, reservoir and estuary samples exhibited lowest dispersion with distances to the median centroid ranging between 0.39 to 0.49. One has to note here that both PERMANOVA and PERMDISP calculate the differences in factors across all PCoA axes. The visualised two axes capture most of the variance in the reservoir and riverine samples, thus they appear to be more variable. However, the variance of terrestrial samples seems to be spread over multiple axes and thus the cumulative dispersion across all dimensions results in higher values.

\subsection*{Divergence of freshwater RNA assemblages}
When DNA and RNA samples were combined, the second PCoA analysis captured in summary 19.85 \% of the dissimilarity variance in the first three axes, deliniating habitat type (PC1), nucleic acid type (PC2) and seasons (PC3)(Fig.3). PERMANOVA analysis indicated significant clustering by habitat type (\textit{F}\textsubscript{12} = 20.65, \textit{R}\textsuperscript{2} = 0.29, \textit{p} = 0.0001), season (\textit{F}\textsubscript{2} = 14.86, \textit{R}\textsuperscript{2} = 0.35, \textit{p} = 0.0001) and nucleic acid type (\textit{F}\textsubscript{1} = 26.07, \textit{R}\textsuperscript{2} = 0.03, \textit{p} = 0.0001). Similarly to the DNA PCoA, homogeneity of dispersion was mostly not fulfilled. According to PERMDISP, dispersion differed by all factorial combinations (\textit{F}\textsubscript{48} = 10.96, \textit{p} < 0.0001), habitat type (\textit{F}\textsubscript{12} = 31.58, \textit{p} = 0.0001) and season (\textit{F}\textsubscript{2} = 44.26, \textit{p} = 0.0001), however, not for solely nucleic acid type (\textit{F}\textsubscript{1} = 1.65, \textit{p} = 0.12).

Overall, the PCoA's first three axes indicate that there is a very substantial seasonality in freshwater samples in both DNA and RNA. A similar seasonal and nucleic acid type dissimilarity to freshwater samples cannot be observed in terrestrially influenced samples. However, it is noteworthy here that there were 29 axes with eigenvalues above 1, together summing up to 43.1 \% of the variance. Visual inspection indicated that a lot of the terrestrially influenced samples' dissimilarity was indeed split upon these additional axes, a clearer separation of soil and soilwater samples for example was observed in axes 6 (1.75\%), 10 (1.12\%), 11 (1.06\%) and 12 (0.99\%)(data not shown). However, apparent seasonality and large differences in nucleic acid type was not observed.

To further explore what is causing DNA and RNA assemblages to diverge across the whole dataset, we compared the pairwise dissimilarity between DNA and RNA of the same sample based on Bray-Curtis ($D_{BC}$) and S{\o}rensen ($D_{S}$) dissimilarity (Fig. 4a). As ($D_{S}$) is solely based on incidences (i.e. presence-absence) we can see when dissimilarities are driven by shuffling in present taxa or differences in abundances. When plotted together, it seems that overall dissimilarity between DNA and RNA is highest in terrestrially influenced samples (e.g. soil, soilwater), however, $D_{BC}$ and $D_{S}$ are fa

\subsection*{Community shifts are driven by all abundance groups}


\section*{Discussion}

 A potential driver of interannual differences are annual variations in hydrology. However, for 2015 and 2016, monthly total precipitation did not substantially vary in spring with 77.5 $\pm$ 5.51 mm in April, 67.4 $\pm$ 4.24 mm in May, 123.2 $\pm$ 13.72 mm in June. However, daily precipitations had a higher variance in June 2016 compared to 2015.

 Criticism has arised to use RNA of the 16S rRNA gene to infer activity of microorganisms out of concern that several biases may obscure adequate interpretations. The main concerns circle around evidence of multiple 16S copy numbers within different microbial genomes (xx) and differing biases introduced between DNA and RNA preparation and sequencing approaches that make DNA:RNA ratios of individual taxa difficult to interpret. Nevertheless, it is a premature decision to completely abandon 16S rRNA RNA approaches before exploring all potential applications.

\subsection*{Data accessibility}
The raw 16S rRNA gene seqeuences, both DNA and cDNA are available at the public NCBI Sequence Read Archive (SRA) under the accession numbers SRxxxxx. The code and meta data that were used to produce this manuscript are available at \url{https://github.com/CarBBAS/xxx} (Zenodo DOI). (Links, accession numbers and DOI will be updated upon acceptance) \\

\section*{Conflict of Interest}
The authors declare no conflict of interest.

\section*{Acknowledgements}
We are especially grateful to Alice H. Parkes, Annick St-Pierre, and Serge Paquet, who maintained and oversaw the La Romaine project over the years. Collection and analysis of all variables would have not been possible without the support from members of the CarBBAS team including great contributions from undergraduate students. Therefore, we would like to thank Felipe Rust, Clara Ruiz-Gonz\`{a}lez, Trista Vick-Majors, Alexandre Ducharme, Roy Nahas, Ryan Hutchins, Marie Laure G\'{e}rardin,  Erin Hotchkiss, Karelle Desrosiers, Martin Demers, Sara Mercier Blais, Julia Jakobsson, Francesca del Giorgio, Brenden Chabot, Sebastian Dugas, and Pascale Ouimet. We would also like to thank Katherine Velghe and Marilyne Robidoux for laboratory assistance, Mario Muscarella for insightful discussions and Robert Ptacnik and Yves Prairie for statistical advice. We also thank xx anonymous reviewers for constructive comments that improved the manuscript. This study is part of the program of the Carbon Biogeochemistry in Boreal Aquatic Systems (CarBBAS) Industrial Research Chair, co-funded by the Natural Science and Engineering Research Council of Canada (NSERC) and Hydro-Qu\'{e}bec.

\newpage
\singlespacing

\bibliography{/media/shared/Documents/University/PhD/BibTEX/LR_miccom_paper}

\newpage
\section*{Figure legends}
Figure 1. \textbf{Location and overview of the La Romaine catchment}. a) Scale and overview of the whole La Romaine catchment. Samples are represented as points. b) Location of the catchment within Canada and Québec. c) Focus on all built reservoirs RO1 (2015), RO2 (2014) and RO3 (2017) and the headwater stream sub-catchment Petite Romaine (PR). \\
\\
Figure 2. \textbf{Microbial community composition gradually changes along a terrestrial-hyrodlogical continuum and diverges between seasons.} PCoA analysis reveals microbial community shifts from terrestrial to freshwater samples. Spring and summer-autumn show distinct paths in multivariate space. Seasons are given in different point shapes, colouring visualises different habitat types. Percentage of variance explained are given in brackets for the first and second axes. \\
\\
Figure 3. \textbf{RNA assemblages diverge from DNA within aquatic habitats, less so in terrestrially influenced habitats.} PCoA analysis inclduing RNA samples. a) Visualisation of first and second axis of PCoA, differentiating habitat type and nucleic acid type, respectively. b) Different view on PCoA analysis using the second and third axis, differentiating nucleic acid type and seasons, respectively. Seasons are given in different point shapes, colouring visualises different habitat types, opacity indicates nucleic acid type. Percentage of variance explained are given in brackets for the first, second and third axes.

\section*{Tables}

\begin{table}

\caption{\label{tab:}Abundance groups}
\centering
\resizebox{\linewidth}{!}{
\begin{tabular}[t]{llr}
\toprule
Abundance groups & Criteria & Categorized OTUs\\
\midrule
Universal abundant & Abundant* in all habitats, never absent & 0\\
Universal moderate & Moderate* in all habitats, never absent & 0\\
Universal rare & Rare* in all habitats, never absent & 0\\
\addlinespace
Abundant & Only abundant observations, can be absent & 1\\
Moderate & Only moderate observations, can be absent & 4\\
Rare & Only rare observations, can be absent & 17459\\
\addlinespace
Specialist & \makecell[l]{Only abundant in one habitat type, \\ never abundant in any other habitats} & 1561\\
\addlinespace
Cosmopolitan & \makecell[l]{Shifts between abundant, moderate and rare, \\but present in all habitats} & 46\\
Shifter & Shifts between abundant, moderate and rare\textsuperscript{\dag} & 342\\
\addlinespace
Upper shifter & \makecell[l]{Shifts in the upper fraction (abundant - moderate) \\of the rank abundance curve\textsuperscript{\dag}} & 755\\
Lower shifter & \makecell[l]{Shifts in the lower fraction (moderate - rare) \\of the rank abundance curve\textsuperscript{\dag}} & 17\\
\bottomrule
\multicolumn{3}{l}{\rule{0pt}{1em}\textsuperscript{*} Abundant: 47 >= CSS reads, Moderate: 5 >= CSS reads < 47, Rare: 0 > CSS reads < 5} \\
\multicolumn{3}{l}{\rule{0pt}{1em}\textsuperscript{\dag} Does not have to be present in all habitats}\\
\end{tabular}}
\end{table}


\newpage

\section*{Figures}
\begin{figure}[!ht]
\centering
\includegraphics[width=12cm]{../Figures/LR_miccom_paper_threemaps.png}
\caption{\textbf{Location and overview of the La Romaine catchment}. a) Scale and overview of the whole La Romaine catchment. Samples are represented as points. b) Location of the catchment within Canada and Québec. c) Focus on all built reservoirs RO1 (2015), RO2 (2014) and RO3 (2017) and the headwater stream sub-catchment Petite Romaine (PR).}
%includegraphics[width=1\textwidth]
\end{figure}

\begin{figure}[!ht]
\centering
\includegraphics[width=16cm]{../../Figures/Final/PCoA_log_DNA_collage.png}
\caption{\textbf{Microbial community composition gradually changes along a terrestrial-hyrodlogical continuum and diverges between seasons.} Overall PCoA analysis (a) has been further explored with focus plots on terrestrial, riverine and estuary samples in panels b, c, d, respectively. Overall, the PCoA reveals microbial community shifts from terrestrial to freshwater samples. Spring and summer-autumn show distinct paths in multivariate space. Percentage of variance explained are given in brackets for the first and second axis.}
%includegraphics[width=1\textwidth]
\end{figure}

\begin{figure}[!ht]
\centering
\includegraphics[width=17cm]{../../Figures/Final/PCoA_all_SampleType.png}
\caption{\textbf{RNA assemblages diverge from DNA within aquatic habitats, less so in terrestrially influenced habitats.} PCoA analysis inclduing RNA samples. a) Visualisation of first and second axis of PCoA, differentiating habitat type and nucleic acid type, respectively. b) Different view on PCoA analysis using the second and third axis, differentiating nucleic acid type and seasons, respectively. Seasons are given in different point shapes, colouring visualises different habitat types, opacity indicates nucleic acid type. Percentage of variance explained are given in brackets for the first, second and third axes.}
%includegraphics[width=1\textwidth]
\end{figure}

\begin{figure}[!ht]
\centering
\includegraphics[width=15cm]{../../Figures/Final/dissimBCS_distBC_distratio.png}
\caption{\textbf{Distance between DNA and RNA samples within PCoA ordination space reveals sections within the continuum that are dominated either by mass or selection processes.} a) Pair-wise dissimilarity between DNA and RNA computed with S{\o}rensen (S) against Bray-Curtis (BS) b) Euclidean distance among DNA and RNA assemblages of the same sample within the first three axes of the PCoA computed with Bray-Curtis dissimilarity (\textit{m}\textsubscript{BC}). c) Ratio between distance calculated based on PCoAs computed with S{\o}rensen (\textit{m}\textsubscript{S}) and Bray-Curtis dissimilarity (\textit{m}\textsubscript{BC}).}
%includegraphics[width=1\textwidth]
\end{figure}

\begin{figure}[!ht]
\centering
\includegraphics[width=15cm]{../../Figures/Final/abgroups_rollmean_violin}
\caption{\textbf{DNA and RNA dissimilarities are caused by most abundance groups} a) Difference in DNA and RNA abundance against $m_{BC}$ along the second axis, which captures RNA divergence. Lines represent roll means with bin n = 10. b) Boxplots and violin plots showing the distribution of species scores and thus OTU optima within PCoA space. The middle line represents the median, lower and upper hinges of boxplots correspond to the 25th and 75th percentiles. Upper and lower whiskers expand to the largest and smallest value, respectively, but no further than 1.5 * the inter-quartile range from the hinge. Outliers that lie beyond the whiskers were removed for visualisation purposes. Violin plots visualise the probability density distribution smoothed by a kernel density estimator.}
%includegraphics[width=1\textwidth]
\end{figure}


\end{document}
